stmt/params_test.go

```diff
@@ -48,50 +48,53 @@ func TestDecodeParamsGetAll(t *testing.T) {
 	tests := []struct {
 		s   string
 		exp []string
-		err bool
+		err error
 	}{
-		{``, nil, false},
-		{` `, nil, false},
-		{` :foo`, []string{`bar`}, false},
-		{` :'foo`, nil, true},
-		{`:'foo'`, []string{`'bar'`}, false},
-		{`:'foo':foo`, []string{`'bar'bar`}, false},
-		{`:'foo':foo:"foo"`, []string{`'bar'bar"bar"`}, false},
-		{`:'foo':foo:foo`, []string{`'bar'barbar`}, false},
-		{` :'foo':foo:foo`, []string{`'bar'barbar`}, false},
-		{` :'foo':yes:foo`, []string{`'bar':yesbar`}, false},
-		{` :foo `, []string{`bar`}, false},
-		{`:foo:foo`, []string{`barbar`}, false},
-		{` :foo:foo `, []string{`barbar`}, false},
-		{`  :foo:foo  `, []string{`barbar`}, false},
-		{`'hello'`, []string{`hello`}, false}, // 14
-		{`  'hello''yes'  `, []string{`hello'yes`}, false},
-		{`  'hello':'yes'  `, []string{`hello:'yes'`}, false},
-		{` :'foo `, nil, true},
-		{` :'foo bar`, nil, true},
-		{` :'foo  bar`, nil, true},
-		{` :'foo  bar `, nil, true},
-		{" `foo", nil, true},
-		{" `foo bar`", []string{"foo bar"}, false},
-		{" `foo  :foo`", []string{"foo  :foo"}, false},
-		{` :'foo':"foo"`, []string{`'bar'"bar"`}, false},
-		{` :'foo' :"foo" `, []string{`'bar'`, `"bar"`}, false},
-		{` :'foo' :"foo"`, []string{`'bar'`, `"bar"`}, false},
-		{` :'foo'  :"foo"`, []string{`'bar'`, `"bar"`}, false},
-		{` :'foo'  :"foo" `, []string{`'bar'`, `"bar"`}, false},
-		{` :'foo'  :"foo"  :foo `, []string{`'bar'`, `"bar"`, `bar`}, false},
-		{` :'foo':foo:"foo" `, []string{`'bar'bar"bar"`}, false}, // 30
-		{` :'foo''yes':'foo' `, []string{`'bar'yes'bar'`}, false},
-		{` :'foo' 'yes' :'foo' `, []string{`'bar'`, `yes`, `'bar'`}, false},
-		{` 'yes':'foo':"foo"'blah''no' "\ntest" `, []string{`yes'bar'"bar"blah'no`, `\ntest`}, false},
+		{``, nil, nil},
```

Comment:
Not really part of this, but a mental note: these tests are getting unwieldy, and should likely be compared directly to `psql`'s output via functionality similar to how the `\crosstabview` test data is generated in `tblfmt`. That is, separate the strings out into a `*gen.go` file, and then execute against `psql` to generate a static test file. This test then should load that from that separate file, and check/ensure it generates the same output as what `psql` generated.

Reply:
I agree, I'll try to add this in a follow up PR.